index.js

```diff
@@ -9,53 +9,16 @@
 
 'use strict';
 
+const webpack = require('webpack'); // eslint-disable-line no-unused-vars
+const EncoreProxy = require('./lib/EncoreProxy');
 const WebpackConfig = require('./lib/WebpackConfig');
 const configGenerator = require('./lib/config-generator');
 const validator = require('./lib/config/validator');
-const prettyError = require('./lib/utils/pretty-error');
-const logger = require('./lib/logger');
 const parseRuntime = require('./lib/config/parse-runtime');
-const chalk = require('chalk');
-const levenshtein = require('fast-levenshtein');
-const fs = require('fs');
-const path = require('path');
+const context = require('./lib/context');
 
-let webpackConfig = null;
-let runtimeConfig = require('./lib/context').runtimeConfig;
-
-function initializeWebpackConfig() {
-    if (runtimeConfig.verbose) {
-        logger.verbose();
-    }
-
-    // Display a warning if webpack is listed as a [dev-]dependency
-    try {
-        const packageInfo = JSON.parse(
-            fs.readFileSync(path.resolve(runtimeConfig.context, 'package.json'))
-        );
-
-        if (packageInfo) {
-            const dependencies = new Set([
-                ...(packageInfo.dependencies ? Object.keys(packageInfo.dependencies) : []),
-                ...(packageInfo.devDependencies ? Object.keys(packageInfo.devDependencies) : []),
-            ]);
-
-            if (dependencies.has('webpack')) {
-                logger.warning('Webpack is already provided by Webpack Encore, also adding it to your package.json file may cause issues.');
-            }
-        }
-    } catch (e) {
-        logger.warning('Could not read package.json file.');
-    }
-
-    webpackConfig = new WebpackConfig(runtimeConfig);
-}
-
-// If runtimeConfig is already set webpackConfig can directly
-// be initialized here.
-if (runtimeConfig) {
-    initializeWebpackConfig();
-}
+let runtimeConfig = context.runtimeConfig;
+let webpackConfig = runtimeConfig ? new WebpackConfig(runtimeConfig, true) : null;
```

Comment:
If there is no runtimeConfig, when is this variable initialized?

Reply:
In `Encore.configureRuntimeEnvironment(...)`

That was already the case before, I basically just replaced calls to `initializeWebpackConfig(...)` by `new WebpackConfig(...)` (since the whole "validation" part is now done by the constructor).